packages/inter-protocol/test/smartWallet/test-psm-integration.js

```diff
@@ -193,11 +193,6 @@ test('want stable (insufficient funds)', async t => {
     'Withdrawal of {"brand":"[Alleged: AUSD brand]","value":"[20000n]"} failed because the purse only contained {"brand":"[Alleged: AUSD brand]","value":"[10000n]"}';
   const status = computedState.offerStatuses.get('insufficientFunds');
   t.is(status?.error, `Error: ${msg}`);
-  /** @type {[PromiseRejectedResult]} */
-  // @ts-expect-error cast
-  const result = status.result;
-  t.is(result[0].status, 'rejected');
```

Comment:
The previous code just happened to call `updateStatus` on a `Promise.all()` that covered this case. We get the same results without the `Promise.all()`, so there's nothing to log at that point. The 'rejected' status is redundant with the error detected just above.

Reply:
This is a breaking change and why `test-oracle-integration` is failing. When this code handles a rejection: https://github.com/Agoric/agoric-sdk/blob/daf4294bfa2ac36b1f6993bd0dc808bd27393fcf/packages/smart-wallet/src/offerWatcher.js#L182 it's not making it to this error handler: https://github.com/Agoric/agoric-sdk/blob/0e25d18804ffc064554d8ff6a813b70246ce7b1d/packages/smart-wallet/src/smartWallet.js#L843-L844

And so it's not getting thrown https://github.com/Agoric/agoric-sdk/blob/0e25d18804ffc064554d8ff6a813b70246ce7b1d/packages/smart-wallet/src/smartWallet.js#L874 to the `executeOffer` caller. 

It's a smell that there are two error handlers which both do `updateStatus({ error: err.toString() });`. Let's see if we can refactor this to have one error handler that throws and `executeOffer()` waits for the offer to complete.